test/e2e/storage/persistent_volumes-volumemode.go

```diff
@@ -0,0 +1,480 @@
+			sc, pod, pv, pvc = createPVTestResource(cs, ns, scConfig, pvConfig, pvcConfig)
+			defer deletePVTestResource(f, cs, ns, sc, pod, pv, pvc)
+
+			By("Checking if persistent volume exists as expected volume mode")
+			checkVolumeModeOfPath(pod, volMode, "/mnt/volume1")
+
+			By("Checking if read/write to persistent volume works properly")
+			checkReadWriteToPath(pod, volMode, "/mnt/volume1")
+		})
+
+		// For block unsupported plugins
+		It("should fail to create pod by failing to mount volume", func() {
+			skipBlockUnsupportTestUnlessUnspported(volMode, isBlockSupported)
+
+			scConfig, pvConfig, pvcConfig := generateConfigsForStaticProvisionPVTest(scName, volBindMode, volMode, pvSource)
+			sc, pod, pv, pvc = createPVTestResourceWithFailure(cs, ns, scConfig, pvConfig, pvcConfig)
+			deletePVTestResource(f, cs, ns, sc, pod, pv, pvc)
+		})
+	}
+
+	AssertRecostructionOnKubeletRestart := func() {
+		// For block supported plugins
+		It("should find volume before and after kubelet restart [Disruptive]", func() {
+			skipBlockSupportTestIfUnsupported(volMode, isBlockSupported)
+
+			scConfig, pvConfig, pvcConfig := generateConfigsForStaticProvisionPVTest(scName, volBindMode, volMode, pvSource)
+			sc, pod, pv, pvc = createPVTestResource(cs, ns, scConfig, pvConfig, pvcConfig)
+			defer deletePVTestResource(f, cs, ns, sc, pod, pv, pvc)
+
+			By("Before kubelet restart, checking if persistent volume exists in pod")
+			checkVolumeModeOfPath(pod, volMode, "/mnt/volume1")
+
+			By("Restarting kubelet")
+			utils.KubeletCommand(utils.KRestart, cs, pod)
+
+			By("Checking events for pod")
+			framework.ExpectNoError(framework.ListPodEvents(cs, ns, pod.Name))
+
+			By("After kubelet restart, checking if persistent volume exists in pod")
+			checkVolumeModeOfPath(pod, volMode, "/mnt/volume1")
+		})
+
+		// For block unsupported plugins
+		// Fail before this test(This should be covered with AssertCreateDeletePodAndReadWriteVolume())
+	}
+
+	AssertVolumeUnmountsFromDeletedPod := func() {
+		// For block supported plugins
+		It("should unmount volume from deleted pod after kubelet restart [Disruptive]", func() {
+			skipBlockSupportTestIfUnsupported(volMode, isBlockSupported)
+
+			scConfig, pvConfig, pvcConfig := generateConfigsForStaticProvisionPVTest(scName, volBindMode, volMode, pvSource)
+			sc, pod, pv, pvc = createPVTestResource(cs, ns, scConfig, pvConfig, pvcConfig)
+			defer deletePVTestResource(f, cs, ns, sc, pod, pv, pvc)
+
+			// TODO(mkimuram): Confirm if this check needs to be different for Block volume
+			utils.TestVolumeUnmountsFromDeletedPodWithForceOption(cs, f, pod, false, false)
+		})
+
+		// For block unsupported plugins
+		// Fail before this test(This should be covered with AssertCreateDeletePodAndReadWriteVolume())
+	}
+
+	verifyAll := func() {
+		AssertCreateDeletePodAndReadWriteVolume()
+		AssertVolumeUnmountsFromDeletedPod()
+		AssertRecostructionOnKubeletRestart()
+	}
+
+	Describe("NFS [Feature:Volumes]", func() {
```

Comment:
This should work without `[Feature:Volumes]`.

Reply:
Confirmed the discussion in https://github.com/kubernetes/kubernetes/pull/43682 . I will remove it.